### DISCUSSION TITLE
How to have a window in a layout display BashExec process?

### DISCUSSION THREAD

**Discussion by Houdiee** [sayanarijit/xplr#735] [Q&A]
I was wondering if it would be possible to get file preview working using  the```bat``` command, where each time a file is focused  it runs Bash as a background process and displays the contents in the layout. Curently, there is only a preview for something [like this](https://xplr.dev/en/awesome-hacks#text-preview-pane) in the docs, which requires to read the file in Lua and return its contents as a string to display in a [CustomParagraph](https://xplr.dev/en/layout#customparagraph). of course the issue with this is that you miss out on syntax highlighting and its kinda limiting if you wanted to do some other stuff as well. I was just wondering if this is possible?

**Comment by sayanarijit** [comment]
It's possible, but xplr is written in sync rust, so it'd be very laggy.
One way is to display files in another window using something like https://github.com/sayanarijit/preview-tabbed.xplr

Or use a tool that makes use of async operations, e.g. https://github.com/sxyazi/yazi

**Reply by Houdiee** [reply]
have you taken a look at this: https://crates.io/crates/tui-termstill unstable though
